src/test/java/seedu/address/model/pool/PooledPassengerContainsKeywordsPredicateTest.java

```diff
@@ -0,0 +1,79 @@
+package seedu.address.model.pool;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.TypicalPools.OFFICEPOOL;
+import static seedu.address.testutil.TypicalPools.WORKPOOL;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+public class PooledPassengerContainsKeywordsPredicateTest {
+
+    @Test
+    public void equals() {
+        List<String> firstPredicateKeywordList = Collections.singletonList("first");
+        List<String> secondPredicateKeywordList = Arrays.asList("first", "second");
+
+        PooledPassengerContainsKeywordsPredicate firstPredicate =
+                new PooledPassengerContainsKeywordsPredicate(firstPredicateKeywordList);
+        PooledPassengerContainsKeywordsPredicate secondPredicate =
+                new PooledPassengerContainsKeywordsPredicate(secondPredicateKeywordList);
+
+        // same object -> returns true
+        assertTrue(firstPredicate.equals(firstPredicate));
+
+        // same values -> returns true
+        PooledPassengerContainsKeywordsPredicate firstPredicateCopy =
+                new PooledPassengerContainsKeywordsPredicate(firstPredicateKeywordList);
+        assertTrue(firstPredicate.equals(firstPredicateCopy));
+
+        // different types -> returns false
+        assertFalse(firstPredicate.equals(1));
+
+        // null -> returns false
+        assertFalse(firstPredicate.equals(null));
+
+        // different passenger -> returns false
+        assertFalse(firstPredicate.equals(secondPredicate));
+    }
+
+    @Test
+    public void test_nameContainsKeywords_returnsTrue() {
+        // One keyword
+        PooledPassengerContainsKeywordsPredicate predicate =
+                new PooledPassengerContainsKeywordsPredicate(Collections.singletonList("Alice"));
```

Comment:
these names could also be taken from the test utils instead of using a magic literal string